resctrl/utils.go

```diff
@@ -0,0 +1,344 @@
+	}
+
+	for _, pid := range pids {
+		threadFiles, err := ioutil.ReadDir(filepath.Join(processPath, pid, processTask))
+		if err != nil {
+			return "", err
+		}
+
+		processThreads, err := getAllProcessThreads(threadFiles)
+		if err != nil {
+			return "", err
+		}
+		for _, thread := range processThreads {
+			err = intelrdt.WriteIntelRdtTasks(monGroupPath, thread)
+			if err != nil {
+				secondError := os.Remove(monGroupPath)
+				if secondError != nil {
+					return "", fmt.Errorf(
+						"coudn't assign pids to %q container monitoring group: %w \n couldn't clear %q monitoring group: %v",
+						containerName, err, containerName, secondError)
+				}
+				return "", fmt.Errorf("coudn't assign pids to %q container monitoring group: %w", containerName, err)
+			}
+		}
+	}
+
+	return monGroupPath, nil
+}
+
+func getPids(containerName string) ([]int, error) {
+	if len(containerName) == 0 {
+		// No container name passed.
+		return nil, fmt.Errorf(noContainerNameError)
+	}
+	pids, err := cgroups.GetAllPids(filepath.Join(pidsPath, containerName))
+	if err != nil {
+		return nil, fmt.Errorf("couldn't obtain pids for %q container: %v", containerName, err)
+	}
+	return pids, nil
+}
+
+func getAllProcessThreads(threadFiles []os.FileInfo) ([]int, error) {
+	processThreads := make([]int, 0)
+	for _, file := range threadFiles {
+		pid, err := strconv.Atoi(file.Name())
+		if err != nil {
+			return nil, fmt.Errorf("couldn't parse %q file: %v", file.Name(), err)
+		}
+		processThreads = append(processThreads, pid)
+	}
+
+	return processThreads, nil
+}
+
+func findControlGroup(pids []string) (string, error) {
+	if len(pids) == 0 {
+		return "", fmt.Errorf(noPidsPassedError)
+	}
+	availablePaths, err := filepath.Glob(filepath.Join(rootResctrl, "*"))
+	if err != nil {
+		return "", err
+	}
+
+	for _, path := range availablePaths {
+		switch path {
+		case filepath.Join(rootResctrl, cpusFileName):
+			continue
+		case filepath.Join(rootResctrl, cpusListFileName):
+			continue
+		case filepath.Join(rootResctrl, modeFileName):
+			continue
+		case filepath.Join(rootResctrl, sizeFileName):
+			continue
+		case filepath.Join(rootResctrl, infoDirName):
+			continue
+		case filepath.Join(rootResctrl, monDataDirName):
+			continue
+		case filepath.Join(rootResctrl, monGroupsDirName):
+			continue
+		case filepath.Join(rootResctrl, schemataFileName):
+			continue
+		case filepath.Join(rootResctrl, tasksFileName):
+			inGroup, err := arePIDsInControlGroup(rootResctrl, pids)
+			if err != nil {
+				return "", err
+			}
+			if inGroup {
+				return rootResctrl, nil
+			}
+		default:
+			inGroup, err := arePIDsInControlGroup(path, pids)
+			if err != nil {
+				return "", err
+			}
+			if inGroup {
+				return path, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("there is no available control group")
+}
+
+func arePIDsInControlGroup(path string, pids []string) (bool, error) {
```

Comment:
non-blocking - please add a comment, that "pids in cgroups" means "all pids must be there", not just one of them